src/mare/compiler/refer.cr

```diff
@@ -102,6 +102,15 @@ module Mare::Compiler::Refer
       @analysis[node] = info
     end
 
+    def touch(ctx, node : AST::Prefix)
+      case node.op.value
+      when "address_of"
+        unless node.term.is_a? AST::Identifier
+          Error.at node.term, "address_of can be applied only to variable"
```

Comment:
Can you add a test to `refer_spec.cr` that demonstrates this error?

If we don't have tests for it I could accidentally break it at any time and not know that I did.

Reply:
Sure